easybuild/easyblocks/g/gcc.py

```diff
@@ -305,6 +325,26 @@ def run_configure_cmd(self, cmd):
         if unknown_options:
             raise EasyBuildError("Unrecognized options found during configure: %s", unknown_options)
 
+    def prepare_step(self, *args, **kwargs):
+        """
+        Check that optional dependencies have been downloaded
+        """
+        super(EB_GCC, self).prepare_step(*args, **kwargs)
+        if self.cfg['withnvptx']:
+            nvptx_dir = glob.glob(os.path.join(self.builddir, 'nvptx-tools-*'))
+            if not nvptx_dir:
```

Comment:
I would store the result to `self.nvptx_dir` so it can be reused later rather than redoing the `glob`, and also ensure there's only one:

```python
nvptx_dir_pattern = os.path.join(self.builddir, 'nvptx-tools-*')
hits = glob.glob(nvptx_dir_pattern)
if len(hits) == 1:
    self.nvptx_dir = hits[0]
else:
    error_msg = "Failed to isolate nvptx-tools directory: %s\n" % hits
    error_msg += "..."
    raise EasyBuildError(error_msg)
```

Likewise for `lld-*` and `llvm-*`, of course.